service/src/main/kotlin/fi/espoo/evaka/document/childdocument/ChildDocumentController.kt

```diff
@@ -78,21 +78,37 @@ class ChildDocumentController(private val accessControl: AccessControl) {
         user: AuthenticatedUser,
         clock: EvakaClock,
         @PathVariable documentId: ChildDocumentId
-    ): ChildDocumentDetails {
+    ): ChildDocumentWithPermittedActions {
         return db.connect { dbc ->
-                dbc.read { tx ->
-                    accessControl.requirePermissionFor(
+            dbc.read { tx ->
+                accessControl.requirePermissionFor(
+                    tx,
+                    user,
+                    clock,
+                    Action.ChildDocument.READ,
+                    documentId
+                )
+
+                val document =
+                    tx.getChildDocument(documentId)?.also {
+                        Audit.ChildDocumentRead.log(targetId = documentId)
```

Comment:
Audit logging should happen outside of the transaction

Reply:
fixed 👍